.github/workflows/CI.yml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on: 
```

Comment:
We will want to run this CI script on:

*   pull requests to `main`
*   Once code is pushed to `main`
*   Daily
*   On Demand

This configuration will do that:
```
on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
  schedule:
    - cron: '0 0 * * *' # Once a day at 12am UTC
  workflow_dispatch:
```